### PR TITLE
01 docker

### PR DESCRIPTION
**Pull request by g0rdonL**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    ports:
+      - 8080:8080
+  nginx:
+    restart: always
+    build:
+      context: ./nginx
+    depends_on:
+      - nodeserver
+    ports:
+      - 80:80
```
Comment (jkotrc) [inline docker-compose.yml:21]: Now we need to include tags for the public network and the private network. To define networks, it's enough to include an empty tag of them:
```
networks:
        front-tier: {}
        back-tier: {}
```
Then include them as necessary:
```
nginx:
    ...
    networks: 
        - front-tier
        - back tier
 nodeserver:
    ...
    networks:
        - back-tier
```


See this very helpful reference: https://github.com/compose-spec/compose-spec/blob/master/spec.md

File: app/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "dependencies": {
+    "express": "^4.17.1",
+    "react": "^17.0.2"
```
Comment (jkotrc) [inline app/package.json:4]: Is it better to use pure react for our webpage? I was under the impression that Nextjs gives us some very helpful features like server side rendering, but we can make do without it?

**Review by jkotrc — CHANGES_REQUESTED** [review]
Some additional public-private network configuration missing in `docker-compose.yml` and a quick question :)

File: app/package.json (hunk 2 above)
Comment (g0rdonL) [reply app/package.json:4]: It depends of if we want it to be treated as a single-page app or not i think. Not sure. AFAIK I think on pure react the routes are treated as "steps" on the same "page" if you will, whereas NextJs will actually treat them as actual URL redirections. Depends how we want to implement the website.

File: docker-compose.yml (hunk 1 above)
Comment (g0rdonL) [reply docker-compose.yml:28]: added [https://github.com/hellofair/ethoWebsite/pull/2/commits/3aba5010700b4029e2002e1aacfbf68ede8260e7](url)

**Pull request merged**